flink-filesystems/flink-gs-fs-hadoop/pom.xml

```diff
@@ -204,20 +251,13 @@ under the License.
 									<pattern>org.apache.flink.runtime.util</pattern>
 									<shadedPattern>org.apache.flink.fs.gs.org.apache.flink.runtime.util</shadedPattern>
 								</relocation>
+								<!-- shade google guava version-->
+								<relocation>
+									<pattern>com.google.common</pattern>
+									<shadedPattern>org.apache.flink.fs.gs.com.google.common</shadedPattern>
```

Comment:
This wouldn't work, right? Because I'm assuming GCS is looking for `com.google.common` and not `org.apache.flink.fs` ?

Reply:
With shading it would rewrite the import:`com.google.common` in classes of module:`flink-gs-fs-hadoop`.

Reply:
Yes but the code from GCS still expects the import to be `com.google.common` because it doesn't know about the shaded class.